lib/model/migrations/20221114-01-explict-dataset-publish.js

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const up = async (db) => {
+  await db.schema.table('entity_defs', (eDef) => {
+    eDef.integer('submissionDefId').alter({ nullable: true });
```

Comment:
In addition to setting this to nullable, I was thinking you could set this onDelete command. This is like my strategy in `./20220121-01-form-cascade-delete.js` but using `.onDelete('set null')` instead of `.onDelete('cascade')`:
```
eDef.dropForeign('submissionDefId');
eDef.foreign('submissionDefId').references('submission_defs.id').onDelete('set null');
```